### PR TITLE
fix(query): expr children with projectset not fold count

### LINKED ISSUE #10826: bug: unnest should not fold count
**Summary**

img_v2_33a415da-1d15-4840-b02a-79ba46ab9b7g

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

img_v2_33a415da-1d15-4840-b02a-79ba46ab9b7g

expr children with projectset not fold count

Closes #10826

### CODE REVIEW

File: tests/sqllogictests/suites/query/02_function/02_0062_function_unnest [hunk 1]
```diff
@@ -369,6 +369,35 @@ drop table t;
...
+4xy
+
+query I
+select count(b) from (select concat(c,'y') as b from (select concat(c::String, 'x')  as c from (select unnest(id) as c from fold_count)));
+----
+4
+
+#ERROR 1105 (HY000): Code: 1006, Text = Unable to get field named "0". Valid fields: [].
+statement error 1006
+select count() from (select unnest(id) from fold_count)
```
Comment (sundy-li) [inline tests/sqllogictests/suites/query/02_function/02_0062_function_unnest:396]: It's error of unnest ?

Comment (andylokandy) [reply tests/sqllogictests/suites/query/02_function/02_0062_function_unnest:396]: the main build returns `1`, which is not a error but still incorrect.

File: tests/sqllogictests/suites/query/02_function/02_0062_function_unnest [hunk 2]
```diff
@@ -373,27 +373,28 @@ statement ok
...
-select count(b) from (select concat(c,'y') as b from (select concat(c::String, 'x')  as c from (select unnest(id) as c from fold_count)));
+select count(b) from (select concat(c,'y') as b from (select concat(c::String, c1)  as c from (select unnest(id) as c, c1 from fold_count)));
 ----
 4
 
-#ERROR 1105 (HY000): Code: 1006, Text = Unable to get field named "0". Valid fields: [].
-statement error 1006
-select count() from (select unnest(id) from fold_count)
+query I
+select count() from (select unnest(id), c1 from fold_count)
```
Comment (sundy-li) [inline tests/sqllogictests/suites/query/02_function/02_0062_function_unnest:395]: I think `c1` could be pruned

**Comment by leiysky** [comment]
This should be fixed by setting the `precise_cardinality` to `None` here:https://github.com/datafuselabs/databend/blob/14e0e4c59d31010f032cd31c5d77b2d2fdd56599/src/query/sql/src/planner/plans/project_set.rs#L51-L61

**Review by leiysky — APPROVED** [review]

File: tests/sqllogictests/suites/query/02_function/02_0062_function_unnest [hunk 3]
```diff
@@ -396,6 +396,36 @@ select count() from (select unnest(id), c1 from fold_count)
 ----
 4
 
+query T
+explain select count() from (select unnest(id), c1 from fold_count)
```
Comment (sundy-li) [inline tests/sqllogictests/suites/query/02_function/02_0062_function_unnest:400]: explain should be inside explain folder.

Comment (leiysky) [reply tests/sqllogictests/suites/query/02_function/02_0062_function_unnest]: The explain results between standalone mode and cluster mode are different, should put this test into `standalone` folder.

File: tests/sqllogictests/suites/mode/standalone/explain/unnest.test [hunk 4]
```diff
@@ -0,0 +1,41 @@
+statement ok
+drop table if_exists fold_count;
```
Comment (sundy-li) [inline tests/sqllogictests/suites/mode/standalone/explain/unnest.test:2]: if exists

**Review by sundy-li — APPROVED** [review]

**Comment by TCeason** [comment]
```
Waiting on databend-query 10 seconds...
... connecting to :3307
... connecting to :3307
... connecting to :3307
... connecting to :3307
... connecting to :3307
... connecting to :3307
... connecting to :3307
... connecting to :3307
... connecting to :3307
... connecting to :3307
Traceback (most recent call last):
  File "/runner/_work/databend/databend/scripts/ci/wait_tcp.py", line 38, in <module>
    tcp_ping(args.port, args.timeout)
  File "/runner/_work/databend/databend/scripts/ci/wait_tcp.py", line 26, in tcp_ping
    raise Exception("fail to connect to :{}".format(port))
Exception: fail to connect to :3307
Databend Query


```

Looks like the ci has some question? I try to re-run it also report this error. cc @BohuTANG

**Comment by BohuTANG** [comment]
> ```
> 
> ```
> 
> 
>     
>   
> 
> Looks like the ci has some question? I try to re-run it also report this error. cc @BohuTANG

Yes, @everpcpc will help us to address it.

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
```
[Diff] (-expected|+actual)
-   ProjectSet
+   EvalScalar
+   ├── expressions: [get(unnest (#1))]
    ├── estimated rows: 10.00
-   ├── set returning functions: unnest([1, 2, 3])
-   └── TableScan
-       ├── table: default.system.numbers
-       ├── read rows: 10
-       ├── read bytes: 80
-       ├── partitions total: 1
-       ├── partitions scanned: 1
-       ├── push downs: [filters: [], limit: NONE]
-       └── estimated rows: 10.00
+   └── ProjectSet
+       ├── estimated rows: 10.00
+       ├── set returning functions: unnest([1, 2, 3])
+       └── TableScan
+           ├── table: default.system.numbers
+           ├── read rows: 10
+           ├── read bytes: 80
+           ├── partitions total: 1
+           ├── partitions scanned: 1
+           ├── push downs: [filters: [], limit: NONE]
+           └── estimated rows: 10.00
at tests/sqllogictests/suites/mode/standalone/explain/project_set.test:10
```

**Pull request merged**